contrib/hyperscan-cmake/CMakeLists.txt

```diff
@@ -1,239 +1,232 @@
-    "${LIBRARY_DIR}/src/rose/rose_build_dedupe.cpp"
-    "${LIBRARY_DIR}/src/rose/rose_build_engine_blob.cpp"
-    "${LIBRARY_DIR}/src/rose/rose_build_exclusive.cpp"
-    "${LIBRARY_DIR}/src/rose/rose_build_groups.cpp"
-    "${LIBRARY_DIR}/src/rose/rose_build_infix.cpp"
-    "${LIBRARY_DIR}/src/rose/rose_build_instructions.cpp"
-    "${LIBRARY_DIR}/src/rose/rose_build_lit_accel.cpp"
-    "${LIBRARY_DIR}/src/rose/rose_build_long_lit.cpp"
-    "${LIBRARY_DIR}/src/rose/rose_build_lookaround.cpp"
-    "${LIBRARY_DIR}/src/rose/rose_build_matchers.cpp"
-    "${LIBRARY_DIR}/src/rose/rose_build_merge.cpp"
-    "${LIBRARY_DIR}/src/rose/rose_build_misc.cpp"
-    "${LIBRARY_DIR}/src/rose/rose_build_program.cpp"
-    "${LIBRARY_DIR}/src/rose/rose_build_role_aliasing.cpp"
-    "${LIBRARY_DIR}/src/rose/rose_build_scatter.cpp"
     "${LIBRARY_DIR}/src/rose/rose_build_width.cpp"
     "${LIBRARY_DIR}/src/rose/rose_in_util.cpp"
     "${LIBRARY_DIR}/src/rose/stream.c"
-    "${LIBRARY_DIR}/src/runtime.c"
-    "${LIBRARY_DIR}/src/scratch.c"
+
     "${LIBRARY_DIR}/src/smallwrite/smallwrite_build.cpp"
+    "${LIBRARY_DIR}/src/smallwrite/smallwrite_dump.cpp"
+
     "${LIBRARY_DIR}/src/som/slot_manager.cpp"
+    "${LIBRARY_DIR}/src/som/slot_manager_dump.cpp"
     "${LIBRARY_DIR}/src/som/som_runtime.c"
     "${LIBRARY_DIR}/src/som/som_stream.c"
-    "${LIBRARY_DIR}/src/stream_compress.c"
+
     "${LIBRARY_DIR}/src/util/alloc.cpp"
     "${LIBRARY_DIR}/src/util/charreach.cpp"
     "${LIBRARY_DIR}/src/util/clique.cpp"
     "${LIBRARY_DIR}/src/util/compile_context.cpp"
     "${LIBRARY_DIR}/src/util/compile_error.cpp"
-    "${LIBRARY_DIR}/src/util/cpuid_flags.c"
     "${LIBRARY_DIR}/src/util/depth.cpp"
+    "${LIBRARY_DIR}/src/util/dump_charclass.cpp"
+    "${LIBRARY_DIR}/src/util/dump_mask.cpp"
     "${LIBRARY_DIR}/src/util/fatbit_build.cpp"
-    "${LIBRARY_DIR}/src/util/multibit_build.cpp"
     "${LIBRARY_DIR}/src/util/multibit.c"
+    "${LIBRARY_DIR}/src/util/multibit_build.cpp"
     "${LIBRARY_DIR}/src/util/report_manager.cpp"
-    "${LIBRARY_DIR}/src/util/simd_utils.c"
     "${LIBRARY_DIR}/src/util/state_compress.c"
     "${LIBRARY_DIR}/src/util/target_info.cpp"
     "${LIBRARY_DIR}/src/util/ue2string.cpp"
-)
 
-add_library (_hyperscan ${SRCS})
+    "${LIBRARY_DIR}/src/util/arch/arm/cpuid_flags.c"
+    "${LIBRARY_DIR}/src/util/arch/x86/cpuid_flags.c"
+    "${LIBRARY_DIR}/src/util/arch/x86/masked_move.c"
 
-target_compile_options (_hyperscan
-    PRIVATE -g0 # Library has too much debug information
-    -mno-avx -mno-avx2 # The library is using dynamic dispatch and is confused if AVX is enabled globally
-    -march=corei7 -O2 -fno-strict-aliasing -fno-omit-frame-pointer -fvisibility=hidden # The options from original build system
-    -fno-sanitize=undefined # Assume the library takes care of itself
-)
-target_include_directories (_hyperscan
-    PRIVATE
-        common
-        "${LIBRARY_DIR}/include"
+    "${LIBRARY_DIR}/src/util/supervector/arch/arm/impl.cpp"
```

Comment:
Most likely we cannot unconditionally compile all three?

Reply:
True. This was just the very first try. The goal here is to compile x86, arm and ppc separately.

By the way, vectorscan and hyperscan diverged quite a bit already. The former also has a new dependency on "Ragel" (http://www.colm.net/open-source/ragel/). I'll check but perhaps we need a fourth-party lib to make the third-party lib work.

Reply:
We can add ragel to contrib at the top level and make vectorscan use it.

Reply:
There are only 2 files of ragel, back then I just invoked ragel on them not to add extra dependency